util/startup: TestStartupFailureRandomRange failed

util/startup.TestStartupFailureRandomRange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11516334?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11516334?buildTab=artifacts#/) on master @ [91907b3e326ae42effdfcac43e0bad0ea2d33d3c](https://github.com/cockroachdb/cockroach/commits/91907b3e326ae42effdfcac43e0bad0ea2d33d3c):


```
=== RUN   TestStartupFailureRandomRange
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange129526242
    test_log_scope.go:81: use -show-logs to present logs inline
    startup_test.go:104: TestStartupFailureRandomRange using seed 7112623644327294821
    test_server_shim.go:131: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    startup_test.go:184: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:184
        	            				github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:105
        	Error:      	Received unexpected error:
        	            	pq: unknown cluster setting 'kv.allocator.load_based_rebalancing'
        	Test:       	TestStartupFailureRandomRange
        	Messages:   	failed to disable load rebalancer
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange129526242
--- FAIL: TestStartupFailureRandomRange (19.34s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupFailureRandomRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30998

util/startup.TestStartupFailureRandomRange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11516335?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11516335?buildTab=artifacts#/) on master @ [91907b3e326ae42effdfcac43e0bad0ea2d33d3c](https://github.com/cockroachdb/cockroach/commits/91907b3e326ae42effdfcac43e0bad0ea2d33d3c):


```
=== RUN   TestStartupFailureRandomRange
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange3348370685
    test_log_scope.go:81: use -show-logs to present logs inline
    startup_test.go:104: TestStartupFailureRandomRange using seed 7122132634678901952
    test_server_shim.go:131: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    startup_test.go:184: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:184
        	            				github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:105
        	Error:      	Received unexpected error:
        	            	pq: unknown cluster setting 'kv.allocator.load_based_rebalancing'
        	Test:       	TestStartupFailureRandomRange
        	Messages:   	failed to disable load rebalancer
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange3348370685
--- FAIL: TestStartupFailureRandomRange (130.67s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupFailureRandomRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

util/startup.TestStartupFailureRandomRange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11530226?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11530226?buildTab=artifacts#/) on master @ [ef9d4266bd4d301e50b3ae35c5582d40ac6e54d3](https://github.com/cockroachdb/cockroach/commits/ef9d4266bd4d301e50b3ae35c5582d40ac6e54d3):


```
=== RUN   TestStartupFailureRandomRange
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange3025388204
    test_log_scope.go:81: use -show-logs to present logs inline
    startup_test.go:104: TestStartupFailureRandomRange using seed 4525597681070339959
    test_server_shim.go:131: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    startup_test.go:184: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:184
        	            				github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:105
        	Error:      	Received unexpected error:
        	            	pq: unknown cluster setting 'kv.allocator.load_based_rebalancing'
        	Test:       	TestStartupFailureRandomRange
        	Messages:   	failed to disable load rebalancer
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange3025388204
--- FAIL: TestStartupFailureRandomRange (17.64s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupFailureRandomRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

util/startup.TestStartupFailureRandomRange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11530227?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11530227?buildTab=artifacts#/) on master @ [ef9d4266bd4d301e50b3ae35c5582d40ac6e54d3](https://github.com/cockroachdb/cockroach/commits/ef9d4266bd4d301e50b3ae35c5582d40ac6e54d3):


```
=== RUN   TestStartupFailureRandomRange
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange3515543457
    test_log_scope.go:81: use -show-logs to present logs inline
    startup_test.go:104: TestStartupFailureRandomRange using seed 4473999317027092664
    test_server_shim.go:131: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    startup_test.go:184: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:184
        	            				github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:105
        	Error:      	Received unexpected error:
        	            	pq: unknown cluster setting 'kv.allocator.load_based_rebalancing'
        	Test:       	TestStartupFailureRandomRange
        	Messages:   	failed to disable load rebalancer
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange3515543457
--- FAIL: TestStartupFailureRandomRange (62.66s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupFailureRandomRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

util/startup.TestStartupFailureRandomRange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11546536?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11546536?buildTab=artifacts#/) on master @ [72636bda49ab5b2ca1b57cbc5cb1d86df7e7ef82](https://github.com/cockroachdb/cockroach/commits/72636bda49ab5b2ca1b57cbc5cb1d86df7e7ef82):


```
=== RUN   TestStartupFailureRandomRange
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange713305105
    test_log_scope.go:81: use -show-logs to present logs inline
    startup_test.go:104: TestStartupFailureRandomRange using seed 649796467497130221
    test_server_shim.go:132: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    startup_test.go:184: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:184
        	            				github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:105
        	Error:      	Received unexpected error:
        	            	pq: unknown cluster setting 'kv.allocator.load_based_rebalancing'
        	Test:       	TestStartupFailureRandomRange
        	Messages:   	failed to disable load rebalancer
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange713305105
--- FAIL: TestStartupFailureRandomRange (19.18s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupFailureRandomRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

util/startup.TestStartupFailureRandomRange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11546537?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11546537?buildTab=artifacts#/) on master @ [72636bda49ab5b2ca1b57cbc5cb1d86df7e7ef82](https://github.com/cockroachdb/cockroach/commits/72636bda49ab5b2ca1b57cbc5cb1d86df7e7ef82):


```
=== RUN   TestStartupFailureRandomRange
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange1660312694
    test_log_scope.go:81: use -show-logs to present logs inline
    startup_test.go:104: TestStartupFailureRandomRange using seed 3205590734743510889
    test_server_shim.go:132: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    startup_test.go:184: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:184
        	            				github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:105
        	Error:      	Received unexpected error:
        	            	pq: unknown cluster setting 'kv.allocator.load_based_rebalancing'
        	Test:       	TestStartupFailureRandomRange
        	Messages:   	failed to disable load rebalancer
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange1660312694
--- FAIL: TestStartupFailureRandomRange (78.08s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupFailureRandomRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

util/startup.TestStartupFailureRandomRange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11561196?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11561196?buildTab=artifacts#/) on master @ [95dcf06ec7d3ba2cea59c7ce608162f2be71681d](https://github.com/cockroachdb/cockroach/commits/95dcf06ec7d3ba2cea59c7ce608162f2be71681d):


```
=== RUN   TestStartupFailureRandomRange
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange1342264893
    test_log_scope.go:81: use -show-logs to present logs inline
    startup_test.go:104: TestStartupFailureRandomRange using seed 6473717585427556849
    test_server_shim.go:132: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    startup_test.go:184: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:184
        	            				github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:105
        	Error:      	Received unexpected error:
        	            	pq: unknown cluster setting 'kv.allocator.load_based_rebalancing'
        	Test:       	TestStartupFailureRandomRange
        	Messages:   	failed to disable load rebalancer
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange1342264893
--- FAIL: TestStartupFailureRandomRange (15.62s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupFailureRandomRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

util/startup.TestStartupFailureRandomRange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11561197?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11561197?buildTab=artifacts#/) on master @ [95dcf06ec7d3ba2cea59c7ce608162f2be71681d](https://github.com/cockroachdb/cockroach/commits/95dcf06ec7d3ba2cea59c7ce608162f2be71681d):


```
=== RUN   TestStartupFailureRandomRange
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange2213770126
    test_log_scope.go:81: use -show-logs to present logs inline
    startup_test.go:104: TestStartupFailureRandomRange using seed 8397055805631991593
    test_server_shim.go:132: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    startup_test.go:184: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:184
        	            				github.com/cockroachdb/cockroach/pkg/util/startup_test/pkg/util/startup/startup_test.go:105
        	Error:      	Received unexpected error:
        	            	pq: unknown cluster setting 'kv.allocator.load_based_rebalancing'
        	Test:       	TestStartupFailureRandomRange
        	Messages:   	failed to disable load rebalancer
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/de1a9605134b8a4c9ee2cfadfe02bbb3/logTestStartupFailureRandomRange2213770126
--- FAIL: TestStartupFailureRandomRange (59.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupFailureRandomRange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/base/test_server_args.go
+++ b/pkg/base/test_server_args.go
@@ -246,7 +246,7 @@ var (
 	// is attempted. This is the default behavior.
 	TestTenantProbabilisticOnly = DefaultTestTenantOptions{testBehavior: ttProb, allowAdditionalTenants: false}
 
-	// TestTenantProbabilisticOnly starts the test under a virtual
+	// TestTenantProbabilistic starts the test under a virtual
 	// cluster on a probabilistic basis. It allows the starting of
 	// additional virtual clusters.
 	TestTenantProbabilistic = DefaultTestTenantOptions{testBehavior: ttProb, allowAdditionalTenants: true}

--- a/pkg/util/startup/startup_test.go
+++ b/pkg/util/startup/startup_test.go
@@ -179,7 +179,7 @@ func runCircuitBreakerTestForKey(
 	require.NoError(t, tc.WaitFor5NodeReplication(), "failed to succeed 5x replication")
 
 	tc.ToggleReplicateQueues(false)
-	c := tc.ServerConn(0)
+	c := tc.Server(0).SystemLayer().SQLConn(t, "")
 	_, err := c.ExecContext(ctx, "set cluster setting kv.allocator.load_based_rebalancing='off'")
 	require.NoError(t, err, "failed to disable load rebalancer")
 	_, err = c.ExecContext(ctx,

